examples/pytorch/infograph/model.py

```diff
@@ -0,0 +1,304 @@
+                                ReLU()
+                                )
+
+        self.jump_con = Linear(in_dim, hid_dim)
+
+    def forward(self, feat):
+        block_out = self.block(feat)
+        jump_out = self.jump_con(feat)
+
+        out = block_out + jump_out
+
+        return out
+
+
+''' Unsupervised Setting '''
+
+class GINEncoder(nn.Module):
+    '''
+    Encoder based on dgl.nn.GINConv &  dgl.nn.SumPooling
+    Parameters
+    -----------
+    in_dim: int
+        Input feature size.
+    hid_dim: int
+        Hidden feature size.
+    n_layer: 
+        Number of GIN layers.
+
+    Functions
+    -----------
+    forward(graph, feat):
+        graph: DGLGraph,
+        feat: Tensor
+            [N * D], node features
+    '''
+
+    def __init__(self, in_dim, hid_dim, n_layer):
+        super(GINEncoder, self).__init__()
+
+        self.n_layer = n_layer
+
+        self.convs = ModuleList()
+        self.bns = ModuleList()
+
+        for i in range(n_layer):
+            if i == 0:
+                n_in = in_dim
+            else:
+                n_in = hid_dim
+            n_out = hid_dim
+            block = Sequential(Linear(n_in, n_out),
+                               ReLU(),
+                               Linear(hid_dim, hid_dim)
+                               )
+
+            conv = GINConv(apply_func = block, aggregator_type = 'sum')
+            bn = BatchNorm1d(hid_dim)
+
+            self.convs.append(conv)
+            self.bns.append(bn)
+
+        # sum pooling
+        self.pool = SumPooling()
+
+    def forward(self, graph, feat):
+
+        xs = []
+        x = feat
+        for i in range(self.n_layer):
+            x = F.relu(self.convs[i](graph, x))
+            x = self.bns[i](x)
+            xs.append(x)
+
+        local_emb = th.cat(xs, 1)                    # patch-level embedding
+        global_emb = self.pool(graph, local_emb)     # graph-level embedding
+
+        return global_emb, local_emb
+
+
+class InfoGraph(nn.Module):
+    r"""
+        InfoGraph model for unsupervised setting
+
+    Parameters
+    -----------
+    in_dim: int
+        Input feature size.
+    hid_dim: int
+        Hidden feature size.
+    n_layer: int   
+        Number of the GNN encoder layers.
+
+    Functions
+    -----------
+    forward(graph):
+        graph: DGLGraph
+
+    """
+
+    def __init__(self, in_dim, hid_dim, n_layer):
+        super(InfoGraph, self).__init__()
+
+        self.in_dim = in_dim
+        self.hid_dim = hid_dim
+
+        self.n_layer = n_layer
+        embedding_dim = hid_dim * n_layer
+
+        self.encoder = GINEncoder(in_dim, hid_dim, n_layer)
+
+        self.local_d = FeedforwardNetwork(embedding_dim, embedding_dim)   # local discriminator (node-level)
+        self.global_d = FeedforwardNetwork(embedding_dim, embedding_dim)  # global discriminator (graph-level)
+
+    def get_embedding(self, graph):
+        # get_embedding function for evaluation the learned embeddings
```

Comment:
To be consistent with the standard DGL practice, it's better to pass `feat = graph.ndata['attr']` to `get_embedding` rather than fetch it inside the function.

Reply:
Do you mean: def get_embedding(graph, feat)?

Reply:
Yes